pilot/cmd/pilot-agent/status/util/stats.go

```diff
@@ -72,7 +78,7 @@ func parseStats(input *bytes.Buffer, stats []*stat) (err error) {
 	}
 	for _, stat := range stats {
 		if !stat.found {
-			err = multierror.Append(err, fmt.Errorf("envoy stat missing: %s", stat.name))
+			*stat.value = 0
```

Comment:
The intent of this error was to help diagnose breaking changes to Envoy version .. does this need to change?

Reply:
Actually, it is the reason for the sidecar not to start in this scenario. Since there is always the invalid cert (does only exist on egress), `update_success` does not exist, but only `update_rejected`. The value is not 0, but the stat is missing, hence this error is triggered and the envoy is never considered ready.

Would it make sense to log a warning here? This would at least help a bit with your intent.

Reply:
@PiotrSikora WDYT?

Reply:
I think it's fine, if those values don't exist and all are equal to 0, then the probe won't get healthy anyway.

Reply:
I'm not entirely sure it's the right fix. What happens on an ingress if the cert is missing ? It might be pretty bad to treat it as healthy.

Why would an egress cert show up in other config ? Maybe the problem is in the config generation - NACKS are also supposed to result in alerts and user notification, it's not a valid state.